### PR TITLE
Add stroke buffer configuration, default to 100

### LINKED ISSUE #300: History size (number of undo levels) should be configurable
- [ ] add a configuration entry
- [ ] add a way to change said configuration entry to the GUI

### PR DESCRIPTION
**Pull request by morinted**
Add buffer configuration, default 100. UI arbitrarily maxes out at 100,000, minimum is set to 1.

Fix #300 

screen shot 2016-02-29 at 4 41 32 pm

### CODE REVIEW

**Comment by Achim63** [comment]
Very welcome as well. Plover even misbehaved majorly when I reached the limit - it started to delete large chunks of text unexpectedly in that case, instead of just stopping to delete anything.

**Comment by benoit-pierre** [comment]
We should take this opportunity to [add a configuration value](https://github.com/benoit-pierre/plover/commit/11b40d64489a7a78d6436212f05ee6d15da5c31d), and partially fix #300. Bonus points if someone add this to the GUI. Not it!

**Comment by morinted** [comment]
All right, all right, I'll do it and amend this commit. By Saturday, okay?

**Comment by benoit-pierre** [comment]
Sure, no hurry!

**Comment by morinted** [comment]
@benoit-pierre UI ui-ified. Probably means reworking my #381 UI, but I'm okay with that.

File: plover/gui/suggestions.py [hunk 1]
```diff
@@ -161,7 +161,8 @@ def lookup_suggestions(self, phrase):
     def show_suggestions(self, suggestion_list):
 
         # Limit history.
-        if len(self.history) == HISTORY_SIZE:
+        undo_levels = self.config.get_undo_levels()
+        while len(self.history) >= undo_levels:
```
Comment (benoit-pierre) [inline plover/gui/suggestions.py]: This code does not work if undo levels is 0, fix [here](https://github.com/morinted/plover/compare/buffer...benoit-pierre:buffer).

Comment (morinted) [inline plover/gui/suggestions.py]: Maybe I'll just limit the minimum to 1 because that's what the shortest dictionary would set it to :)

**Comment by morinted** [comment]
Threw in the assertion, changed the minimum undo length to 1.

**Comment by morinted** [comment]
@benoit-pierre Using the exception caused problems because Plover wouldn't finish initializing, I've just opted for a warning notification instead. How's that?

**Comment by morinted** [comment]
Updated and squashed, right now an invalid configuration (< 1) causes Plover to crash on startup.

**Pull request merged**